### PR TITLE
docs: add waagent manpage

### PR DESCRIPTION
**Pull request by litian1992**
Add a man page for waagent. Addresses issue #3399.

### CODE REVIEW

**Comment by litian1992** [comment]
@microsoft-github-policy-service agree company="Red Hat"

File: doc/man/waagent.1 [hunk 1]
```diff
@@ -0,0 +1,130 @@
+.TH WAAGENT 1 "June 2025" "Azure Linux Agent" "System Administration"
+.SH NAME
+waagent \- Azure Linux Agent for provisioning and managing Linux VMs on Microsoft Azure
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
waagent \- Azure Linux VM Agent
```

File: doc/man/waagent.1 [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+.SH SYNOPSIS
+.B waagent
+[-verbose] [-force] [-help] [\fIOPTION\fR]...
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
[-verbose] [-force] [-help] [\fISUBCOMMAND\fR]...
```

File: doc/man/waagent.1 [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+.SH DESCRIPTION
+The Azure Linux Agent (waagent) manages virtual machine interaction with the Azure fabric controller.
+
+It handles provisioning, configuration, and communication tasks such as:
```
Comment (maddieford) [inline doc/man/waagent.1]: I would remove this and the list below and replace it with:

```suggestion
Most subcommands are not meant to be run directly by the user. However, some subcommands may be useful for debugging (such as collect-logs, version, and show-configuration) and deprovisioning. 
```

File: doc/man/waagent.1 [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+\- Creating and formatting the root device  
+\- Setting hostname and networking  
+\- Reporting VM status  
+\- Managing user accounts and SSH keys  
+\- Handling certificates  
+\- Supporting diagnostics and extensions  
+
+.SH OPTIONS
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
.SH SUBCOMMANDS
```

File: doc/man/waagent.1 [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+.TP
+\fB-collect-logs\fR
+Collect waagent logs and zip it to /var/lib/waagent/logcollector/logs.zip. Default to normal mode. Use \fB-full\fR to collect full logs.
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
Runs the log collector utility that collects relevant agent logs for debugging and stores them in the agent folder on disk. Exact location will be shown when run. Use flag \fB-full\fR for more exhaustive log collection.
```

File: doc/man/waagent.1 [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+\fB-configuration-path FILE\fR
+Used together with \fB-start\fR or \fB-daemon\fR to specify configuration file. Default to /etc/waagent.conf.
+
+.TP
+\fB-daemon -start\fR
+Run waagent as a daemon in background.
+
+.TP
+\fB-deprovision\fR
+Remove machine-specific information from the VM and prepare for image capture. Does not remove the provisioning configuration.
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
Attempt to clean the system and make it suitable for re-provisioning. WARNING: Deprovision does not guarantee that the image is cleared of all sensitive information and suitable for redistribution.
```

File: doc/man/waagent.1 [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+.TP
+\fB-deprovision+user\fR
+Same as \fB-deprovision\fR, but also removes the last provisioned user account.
+
+.TP
+\fB-register-service\fR
+Register waagent as a service - waagent.service - and enable it.
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
Register waagent as a service and enable it.
```

File: doc/man/waagent.1 [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+.TP
+\fB-help\fR
+Display usage information.
+
+.TP
+\fB-version\fR
+Show the current version of the agent.
+
+.SH CONFIGURATION
+The agent is configured via the file by default:
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
The agent is configured via this file by default:
```

File: doc/man/waagent.1 [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+Example entries:
+.RS
+Provisioning.Enabled=y  
+ResourceDisk.Format=y  
+ResourceDisk.MountPoint=/mnt/resource  
+RSA.KeyLength=2048  
+Logs.Verbose=y
+.RE
+
+.SH FILES
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
.SH FILES AND DIRECTORIES
```

File: doc/man/waagent.1 [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+State files and provisioning artifacts.
+
+.TP
+\fB/var/log/waagent.log\fR
+Agent log file.
+
+.SH SERVICES
+On systemd systems, the agent runs as:
+.RS
+.B systemctl start waagent.service
```
Comment (maddieford) [inline doc/man/waagent.1]:
```suggestion
.B systemctl start waagent.service  
```
depending on the distro, this may be walinuxagent.service

File: doc/man/waagent.1 [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+.B systemctl start waagent.service  
+.B systemctl enable waagent.service  
+.RE
+
+.SH EXIT STATUS
+Zero on success, non-zero on error.
+
+.SH EXAMPLES
+.TP
+Provision a newly created VM:
```
Comment (maddieford) [inline doc/man/waagent.1]: Let's remove this example, we don't recommend the user do this

**Review by maddieford — COMMENTED** [review]
Do you plan to update setup.py to set this file for rhel distros (or will it get copied to the machine via a different distribution method)? 

If setup.py makes sense to you, we can update setup.py to create the file at this path for rhel distros: '/usr/share/man/man1'. Does that look right to you?

**Comment by maddieford** [comment]
Thank you for the contribution @litian1992! I left a few comments on the PR

**Comment by litian1992** [comment]
> Do you plan to update setup.py to set this file for rhel distros (or will it get copied to the machine via a different distribution method)?
> 
> If setup.py makes sense to you, we can update setup.py to create the file at this path for rhel distros: '/usr/share/man/man1'. Does that look right to you?

Thank you for your advice. I added a handler in setup.py blindly for man pages. I don't know the proper way of doing it. But I'm happy to learn. Let me know what you think please.

**Review by narrieta — APPROVED** [review]
Thanks for your contribution. Could you base the PR on the develop branch?

**Comment by litian1992** [comment]
> Thanks for your contribution. Could you base the PR on the develop branch?

Done. Can you re-approve please. Thank you.

**Review by narrieta — APPROVED** [review]
Thank you!

**Comment by litian1992** [comment]
> Thank you @litian1992 ! I approved, but the unit tests aren't running. You may need to push an empty commit to force the unit tests to run `git commit --allow-empty -m "Force execution of unit tests"`

Thank you. Done.
